docs/source/onnx/hotwords/index.rst

```diff
@@ -0,0 +1,919 @@
+
+
+    For models trained on ``cjkchar`` it looks like:
+
+     .. code-block::
+
+       语 音 识 别
+       深 度 学 习
+
+    For models trained on ``bpe`` (English like languages) it looks like:
+
+     .. code-block::
+
+      ▁SP E E CH ▁RE C O G N ITION
+      ▁DE E P ▁LE AR N ING
+
+    For models trained on ``cjkchar+bpe`` it looks like:
+
+     .. code-block::
+
+      ▁SP E E CH 识 别
+      ▁SP E E CH ▁RE CO G N ITION    
+      深 度 学 习
+
+  - ``hotwords-score``
+
+    The boosting score for each matched token.
+
+     .. note::
+
+       We match the hotwords at token level, so the ``hotwords-score`` is applied
+       at token level.
+
+
+We provide a command line tool to convert hotwords to tokens, you can see the
+usage as follows:
+
+.. code-block::
+
+   sherpa-onnx text2token --help
+   Usage: sherpa-onnx text2token [OPTIONS] INPUT OUTPUT
+   
+   Options:
+     --tokens TEXT       The path to tokens.txt.
+     --tokens-type TEXT  The type of modeling units, should be cjkchar, bpe or
+                         cjkchar+bpe
+     --bpe-model TEXT    The path to bpe.model.
+     --help              Show this message and exit.
+
+The tool has three options:
+
+  - ``tokens``
+
+    The file path of the tokens.txt, you can find tokens.txt in any of our
+    pretrained model.
+
+  - ``tokens-type``
+
+    The modeling unit used to train the models. Three kinds
+    of ``tokens_type`` are supported now, ``cjkchar``, ``bpe`` and ``cjkchar+bpe``.
+    The ``tokens-type`` tells the systems how to encode ``hotwords`` into tokens.
+
+  - ``bpe-model``
+
+    The file path of the bpe model used to generate the ``tokens.txt``, it is also
+    used to encode the ``hotwords``.
+    Only used when ``tokens-type`` is ``bpe`` or ``cjkchar+bpe``.
+
+
+The main difference of using hotwords feature is about the modeling units (i.e. tokens_type).
+The following shows how to use it for different modeling units.
+
+.. hint::
+
+   You can use any transducer models here `<https://k2-fsa.github.io/sherpa/onnx/pretrained_models/index.html>`_,
+   we just choose three of them randomly for the following examples.
+
+
+.. note::
+
+   In the following example, we use a non-streaming model, if you are using a
+   streaming model, you should use ``sherpa-onnx``. ``sherpa-onnx-alsa``,
+   ``sherpa-onnx-microphone``, ``sherpa-onnx-microphone-offline``,
+   ``sherpa-onnx-online-websocket-server`` and ``sherpa-onnx-offline-websocket-server``
+   all support hotwords.
+
+
+Modeling unit is bpe
+^^^^^^^^^^^^^^^^^^^^
+
+**Download the model**
+
+.. code-block:: bash
+
+   cd /path/to/sherpa-onnx
+
+   GIT_LFS_SKIP_SMUDGE=1 git clone https://huggingface.co/csukuangfj/sherpa-onnx-zipformer-en-2023-04-01
+   cd sherpa-onnx-zipformer-en-2023-04-01
+   git lfs pull --include "*.onnx"
+   git lfs pull --include "bpe.model"
+   cd ..
+   ln -s sherpa-onnx-zipformer-en-2023-04-01 exp
+
+Convert the hotwords into tokens
+
+.. code-block::
+
+   sherpa-onnx text2token \
```

Comment:
`sherpa-onnx` is also a binary.
There would be a conflict for such a design.

Reply:
can we change to `sherpaonnx`? or `sherpa-onnx-cli` ?

Reply:
In that case, I will make it a command line tool and script in the repo, I think I once asked you whether people is able to use sherpa-onnx without cloneing our repo, you said yes. So I think we should make it a command tool.